src/test/java/com/spotify/folsom/IntegrationTest.java

```diff
@@ -118,6 +118,8 @@ public static void tearDownClass() throws Exception {
 
   @Before
   public void setUp() throws Exception {
+    assumeTrue(0 == Utils.getGlobalConnectionCount());
```

Comment:
Should this really be an assumption (i.e ignoring the test if not true), wouldn't we rather want the test to fail in that case as it would highlight something being broken?

Reply:
True I suppose, but then something else should have failed earlier and we can fast fail the rest of the tests.

Reply:
But... I guess we would fastfail anyway with assert - i'll fix